lib/magma/attributes/file.rb

```diff
@@ -76,5 +79,10 @@ def filename(record_name, path)
       ext = '.dat' if ext.empty?
       "#{@model.model_name}-#{record_name}-#{@name}#{ext}"
     end
+
+    def temporary_filepath
+      Magma.instance.storage.upload_url(
+        @model.project_name, "tmp/#{SecureRandom.uuid}")
```

Comment:
Rather than using SecureRandom.uuid here you might prefer Magma.instance.sign.uid, which is usually how I generate random hex IDs in Etna apps; I prefer to centralize all of the cryptographic stuff in that place so all our potential mistakes are in one place.